### PR TITLE
Introduce MAP and JSON_STRING builtin functions

### PR DESCRIPTION
**Pull request by Sxnan**
## What is the purpose of the change

Introduce MAP and JSON_STRING builtin functions.

## Brief change log

- Introduce MAP and JSON_STRING builtin functions

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API: yes

## Documentation

  - Does this pull request introduce a new feature? yes
  - If yes, how is the feature documented? docs

### CODE REVIEW

File: python/feathub/feature_views/transforms/tests/test_expression_transform.py [hunk 1]
```diff
@@ -340,3 +340,113 @@ def test_duplicate_feature_name_with_source(self):
...
+                    name="row_map",
+                    transform="MAP('name', name, "
+                    "'cost', JSON_STRING(cost), "
+                    "'distance', JSON_STRING(distance), "
+                    "'time', time)",
+                ),
+                Feature(
+                    name="cost_distance_map",
+                    transform="MAP('cost', cost, 'distance', distance)",
+                ),
```
Comment (yunfengzhou-hub) [inline python/feathub/feature_views/transforms/tests/test_expression_transform.py:362]: Could you please also add a Feature whose `transform="MAP()"`? Zero arguments is also an even number of arguments.

**Review by yunfengzhou-hub — COMMENTED** [review]
Thanks for the PR. Left some comments as below.

File: python/feathub/feature_views/transforms/tests/test_expression_transform.py (hunk 1 above)
Comment (Sxnan) [reply python/feathub/feature_views/transforms/tests/test_expression_transform.py:362]: The problem with `MAP()` is that we cannot derive the type information of the result. Given that we don't see the use case at the moment, I think it is simpler that the `MAP` function requires at least two arguments.

**Comment by Sxnan** [comment]
@yunfengzhou-hub Thanks for your review! I updated the PR accordingly. Can you take another look?

**Comment by lindong28** [comment]
Thanks for the PR. LGTM.

**Pull request merged**